great_expectations/datasource/fluent/interfaces.py

```diff
@@ -1324,10 +1325,16 @@ def metrics_calculator_result_to_metric_result(
             if metric_name in raw_metrics:
                 MetricType = MetaMetric.get_registered_metric_class_from_metric_name(metric_name)
                 MetricResultType = MetricType.get_metric_result_type()
+                metric_configuration_id = raw_metrics[metric_name][0]
+                value = raw_metrics[metric_name][1]
+                # "condition" metrics return the domain and value kwargs
+                # we just want the value, which is the first item in the tuple
+                if metric_name.endswith(MetricNameSuffix.CONDITION) and isinstance(value, tuple):
+                    value = value[0]
```

Comment:
nit: I'd move this logic into a helper method, eg `parse_metric_value`.

I make make an equivalent one for `parse_metric_config_id`.

These seem like lower level details than this method reads.

Reply:
I think I understand the ask here. I refactored some of this into 2 helper methods and added `NamedTuples` to make it clearer what's going on.